core/src/main/java/org/elasticsearch/bootstrap/BootstrapCLIParser.java

```diff
@@ -0,0 +1,168 @@
+        super(CONFIG, terminal);
+    }
+
+    @Override
+    protected Command parse(String cmdName, CommandLine cli) throws Exception {
+        switch (cmdName.toLowerCase(Locale.ROOT)) {
+            case Start.NAME:
+                return Start.parse(terminal, cli);
+            case Version.NAME:
+                return Version.parse(terminal, cli);
+            default:
+                assert false : "should never get here, if the user enters an unknown command, an error message should be shown before parse is called";
+                return null;
+        }
+    }
+
+    static class Version extends CliTool.Command {
+
+        private static final String NAME = "version";
+
+        private static final CliToolConfig.Cmd CMD = cmd(NAME, Version.class).build();
+
+        public static Command parse(Terminal terminal, CommandLine cli) {
+            return new Version(terminal);
+        }
+
+        public Version(Terminal terminal) {
+            super(terminal);
+        }
+
+        @Override
+        public ExitStatus execute(Settings settings, Environment env) throws Exception {
+            terminal.println("Version: %s, Build: %s/%s, JVM: %s", org.elasticsearch.Version.CURRENT, Build.CURRENT.hashShort(), Build.CURRENT.timestamp(), JvmInfo.jvmInfo().version());
+            return ExitStatus.OK_AND_EXIT;
+        }
+    }
+
+    static class Start extends CliTool.Command {
+
+        private static final String NAME = "start";
+
+        private static final CliToolConfig.Cmd CMD = cmd(NAME, Start.class)
+                .options(
+                        optionBuilder("d", "daemonize").hasArg(false).required(false),
+                        optionBuilder("p", "pidfile").hasArg(true).required(false),
+                        optionBuilder("V", "version").hasArg(false).required(false),
+                        Option.builder("D").argName("property=value").desc("use value for given property").valueSeparator('=').numberOfArgs(2).argName("D")
+                )
+                .lenient(true) // needed to parse the voodoo opts I guess
+                .build();
+
+        public static Command parse(Terminal terminal, CommandLine cli) {
+            if (cli.hasOption("V")) {
+                return Version.parse(terminal, cli);
+            }
+
+            System.setProperty("es.foreground", "true");
+            if (cli.hasOption("d")) {
+                System.clearProperty("es.foreground");
+            }
+
+            String pidFile = cli.getOptionValue("pidfile");
+            if (!Strings.isNullOrEmpty(pidFile)) {
+                System.setProperty("es.pidfile", pidFile);
+            }
+
+            if (cli.hasOption("D")) {
+                Properties properties = cli.getOptionProperties("D");
+                for (Map.Entry<Object, Object> entry : properties.entrySet()) {
+                    System.setProperty("es." + entry.getKey().toString(), entry.getValue().toString());
```

Comment:
We already prepend `es.`.. see [here](https://github.com/elastic/elasticsearch/blob/master/core/bin/elasticsearch#L198)